tests/test_models.py

```diff
@@ -34,9 +34,53 @@ def test_infeasible_model():
         )
     )
     m = solph.Model(es)
-    with warnings.catch_warnings(record=True) as w:
-        m.solve(solver="cbc")
-        assert "Optimization ended with status" in str(w[0].message)
+    with warnings.catch_warnings(record=False) as w:
+        m.solve(solver="cbc", allow_nonoptimal=True)
+        assert "The solver did not return an optimal solution" in str(
+            w[0].message
+        )
```

Comment:
IMHO, it makes sense to also switch to modern pytest usage.

```suggestion
        with pytest.warns(UserWarning, match="The solver did not return an optimal solution"):
            m.solve(solver="cbc", allow_nonoptimal=True)
```